Google.PowerShell/BigQuery/BqTabledata.cs

```diff
@@ -230,4 +231,206 @@ protected override void EndProcessing()
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public WriteDisposition WriteMode { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The number of malformed rows that the request will ignore before throwing an error. 
+        /// This value is zero if not specified.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public int MaxBadRecords { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// CSV ONLY: Allows insertion of rows with fields that are not in the schema, ignoring the extra fields.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter AllowUnknownFields { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// CSV ONLY: Allows insertion of rows that are missing trailing optional columns.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter AllowJaggedRows { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// CSV ONLY: Allows quoted data sections to contain newlines
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter AllowQuotedNewlines { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// CSV ONLY: Separator between fields in the data.  Default value is comma (,).
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public string FieldDelimiter { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// CSV ONLY: Value used to quote data sections.  Default value is double-quote (").
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public string Quote { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// CSV ONLY: The number of rows to skip from the input file. (Usually used for headers.)
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public int SkipLeadingRows { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            Stream fileInput;
+            try
+            {
+                fileInput = new StreamReader(Filename).BaseStream;
+            }
+            catch (Exception ex)
+            {
+                WriteError(new ErrorRecord(ex,
+                    $"Error while reading file {Filename}.",
+                    ErrorCategory.ReadError, Filename));
+                return;
+            }
+            var projectReference = new ProjectReference();
+            projectReference.ProjectId = InputObject.ProjectId;
+            var Client = new BigQueryClientImpl(projectReference, Service);
+            BigQueryJob bqj = null;
```

Comment:
I prefer to leave this uninitialized. That way, I get a compile time check that bqj has been initialized in every case statement.